functions/snippets/src/test/java/SnippetsTests.java

```diff
@@ -179,6 +179,15 @@ public void filesTest() throws IOException {
     assertThat(responseOut.toString(), containsString("Files:"));
   }
 
+  @Test
+  public void logEntry() throws IOException {
+    LogEntry.PubSubMessage message = new LogEntry.PubSubMessage();
+    message.data = "data";
```

Comment:
It's probably best to use GSON to test this, since that seems to be what they are using in the functions environment. This will make the test more like the actual environment. 

Something like:
```java
PubSubMessage message = gson.("{\"data\":\"data\",\"messageId\":\"id\"}", PubSubMessage.class)
new LogEntry().helloPubSub(message);
```
If GSON isn't a dependency already, you can add it and scope to `test`. 

Additionally, according to the [gson user guide](https://sites.google.com/site/gson/gson-user-guide), it looks like GSON will just setting missing fields to null.